### ISSUE TITLE
v6.14.0+ breakes existing model types

### ISSUE THREAD

**Issue by snitin315** [sequelize/sequelize#14129]
## Issue Creation Checklist

[ ] I have read the [contribution guidelines](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)

### SSCCE

**Here is the link to the SSCCE for this issue: https://github.com/opencollective/opencollective-api/pull/7142

```js
// You can delete this code block if you have included a link to your SSCCE above!
import { PaypalProductCreateAttributes } from './PaypalProduct';

interface PaypalPlanAttributes {
  id: string;
  ProductId: string;
  currency: string;
  interval: string;
  amount: number;
  createdAt: Date;
  updatedAt: Date;
  deletedAt: Date;
}

interface PaypalPlanCommonCreateAttributes {
  id: string;
  currency: string;
  interval: string;
  amount: number;
}

interface PaypalPlanCreateWithProductIdAttributes extends PaypalPlanCommonCreateAttributes {
  ProductId?: string;
}

interface PaypalPlanCreateWithProductAttributes extends PaypalPlanCommonCreateAttributes {
  product?: PaypalProductCreateAttributes;
}

type PaypalPlanCreateAttributes = PaypalPlanCreateWithProductIdAttributes | PaypalPlanCreateWithProductAttributes;


class PaypalPlan extends Model<PaypalPlanAttributes, PaypalPlanCreateAttributes> implements PaypalPlanAttributes {
  id: string;
  ProductId: string;
  currency: string;
  interval: string;
  amount: number;
  createdAt: Date;
  updatedAt: Date;
  deletedAt: Date;
  constructor(...args) {
    super(...args);
    restoreSequelizeAttributesOnClass(new.target, this);
  }
}
```

### What do you expect to happen?

No typescript errors for v6.13.0 and before.

### What is actually happening?

<img width="1135" alt="Screenshot 2022-02-16 at 6 53 13 AM" src="https://user-images.githubusercontent.com/46647141/154387973-ec7e0eb6-73be-4c4e-92d3-2c6a58042b92.png">

### Additional context

Using intersection instead of union is fixing it, but we do not want an intersection here.

```diff
- type PaypalPlanCreateAttributes = PaypalPlanCreateWithProductIdAttributes | PaypalPlanCreateWithProductAttributes;
+ type PaypalPlanCreateAttributes = PaypalPlanCreateWithProductIdAttributes & PaypalPlanCreateWithProductAttributes;
```

### Environment

- Sequelize version: XXX 
- Node.js version: XXX 
- If TypeScript related: TypeScript version: XXX 

### How does this problem relate to dialects?

- [ ] I think this problem happens regardless of the dialect.
- [ ] I think this problem happens only for the following dialect(s): 
- [ ] I don't know, I was using PUT-YOUR-DIALECT-HERE, with connector library version XXX and database version XXX

### Would you be willing to resolve this issue by submitting a Pull Request?

- [ ] Yes, I have the time and I know how to start.
- [x] Yes, I have the time but I don't know how to start, I would need guidance.
- [ ] No, I don't have the time, although I believe I could do it if I had the time...
- [ ] No, I don't have the time and I wouldn't even know how to start.

**Comment by snitin315** [comment]
Fixed in #14146

**Comment by ephys** [comment]
Happy to hear it worked 🎉
